src/BitMapViewer.cpp

```diff
@@ -177,31 +177,25 @@ void BitMapViewer::refresh()
 void BitMapViewer::on_screenMode_currentIndexChanged(int index)
 {
 	screenMod = screenMode->itemText(index).toInt();
-	printf("\nnew screenMod: %i\n", screenMod);
-	//change then number of visibe pages when this changes
+	pageSize = screenMod >= 7 ? 65536 : 32768;
+
+	// change then number of visibe pages when this changes
 	imageWidget->setVramAddress(0);
+
 	// make sure that we are decoding new mode from page 0, imagine viewing
 	// page 3 of screen5 and then switching to screen 8 without changing the
 	// starting vram address....
 	imageWidget->setScreenMode(screenMod);
 	setPages();
 }
 
+static const char* pageNames[] = { "0", "1", "2", "3", "E0", "E1" };
+
 void BitMapViewer::setPages()
 {
-	int oldIndex = currentPage->currentIndex();
-
 	currentPage->clear();
-	currentPage->insertItem(0, "0");
-	currentPage->insertItem(1, "1");
-
-	if (screenMod < 7 && VDPDataStore::instance().getVRAMSize() > 0x10000) {
-		currentPage->insertItem(2, "2");
-		currentPage->insertItem(3, "3");
-	}
-
-	if (!useVDP && oldIndex < currentPage->count()) {
-		currentPage->setCurrentIndex(oldIndex);
```

Comment:
The old code tried to restore the old index. You don't do that in the new version.
I don't know enough about Qt to know if that wasn't needed in the old version, or you forgot to do it in the new version? Or maybe you intentionally removed this behavior? (And then it always selects the 1st item?)